### PR TITLE
Implemented `TorqueActuator` in `sympy.physics.mechanics`

### PR DESCRIPTION
**Pull request by brocksam**
#### References to other Issues or PRs

Forms part of the CZI biomechanics work program described in https://github.com/sympy/sympy/issues/24240.

Extends specifically #24966 (but is also related to #24914 and #25068).

#### Brief description of what is fixed or changed
This PR implements a new actuator class, `TorqueActuator`, which is the second concrete class of `ActuatorBase`. It represents an actuator that can applied torque to a pair of reference frames, with a (equal and opposite) reaction torque being applied to the second of the pair. 

I have not added any documentation beyond docstrings, nor have I added the docstrings to the online documentation yet. For the reasoning above this would likely need significant rewriting after any API changes. Documentation will be contributed in a future PR when the module is moved to `sympy/physics/mechanics/actuator.py`.

#### Other comments
I have specifically left the release notes entry empty. These will be added in the PR that moves the module to `sympy/physics/mechanics/actuator.py`.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Review by tjstienstra — COMMENTED** [review]
Nice improvement

File: sympy/physics/mechanics/_actuator.py [hunk 1]
```diff
@@ -249,3 +263,273 @@ def to_loads(self) -> list[tuple[Point, Vector]]:
...
+            which the torque actuator will act. The pair of bodies acted upon
+            by the torque actuator are the parent and child bodies of the pin
+            joint, with the child acting as the reaction body. The pin joint's
+            axis is used as the axis about which the torque actuator will apply
+            its torque.
+
+        """
+        if not isinstance(pin_joint, PinJoint):
+            msg = (
+                f'Value {repr(pin_joint)} passed to `pin_joint` was of type '
```
[the remark this answers is not in the document]
Comment (brocksam) [reply sympy/physics/mechanics/_actuator.py:412]: I think this is correct as is as it is referring to the `pin_joint` parameter.

File: sympy/physics/mechanics/_actuator.py [hunk 2]
```diff
@@ -249,3 +263,273 @@ def to_loads(self) -> list[tuple[Point, Vector]]:
...
+                f'{type(pin_joint)}, must be {PinJoint}.'
+            )
+            raise TypeError(msg)
+        axis = pin_joint.joint_axis
+        body_1 = pin_joint.parent
+        body_2 = pin_joint.child
+        return cls(torque, axis, body_1, body_2)
```
[the remark this answers is not in the document]
Comment (brocksam) [reply sympy/physics/mechanics/_actuator.py:419]: Currently this is only compatible with `RigidBody`s as I thought the `Particle`s don't have frames so it didn't make sense for them to be targets for a torque actuator. If we want to also support `Particle`s then I can change this.

**Comment by brocksam** [comment]
The current implementation requires two bodies to always be specified as a torque actuator must act on a pair of bodies to do useful work. An alternative would be to replace the `*frames` parameter with say `target_frame` and `reaction_frame` and allow the reaction frame to be optional allowing torque actuators to only operate on a single frame (with the assumption being that is no reaction frame is supplied then it is an infinite-inertia Newtonian frame that is acting as that second reacting frame).

File: sympy/physics/mechanics/_actuator.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (TJStienstra) [reply sympy/physics/mechanics/_actuator.py]: Using the interframes, is indeed a correct solution, as those are fixed w.r.t. the bodies.

**Review by tjstienstra — APPROVED** [review]
LGTM

**Pull request merged**